apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -83,23 +87,48 @@ const isUnder1000Chars = (value) => {
   }
 };
 
-const fieldValidationErrors = {
-  primaryEmail: isEmail,
-  secondaryEmail: isEmail,
-  principalEmail: isEmail,
-  phoneNumber: isPhoneNumber,
-  whyCsIsImportant: isUnder1000Chars,
-  whatTeachingSteps: isUnder1000Chars,
-};
-
 const TeacherApplication = React.createClass({
-
   propTypes: {
     regionalPartnerGroup: React.PropTypes.number,
     regionalPartnerName: React.PropTypes.string,
     workshopDays: React.PropTypes.string,
     schoolDistrictData: React.PropTypes.object.isRequired,
-    districtErrorMessageHandler: React.PropTypes.func.isRequired
+    districtErrorMessageHandler: React.PropTypes.func.isRequired,
+    accountEmail: React.PropTypes.string,
+    hashedAccountEmail: React.PropTypes.string.isRequired
+  },
+
+  componentWillMount() {
+    // Validate primary email is a valid email string, and that it matches the account email.
+    // In case it's an accidental student account (which will be converted to teacher on submit)
+    // compare with the hashed email.
+    const validatePrimaryEmail = (value) => {
+      const isEmailError = isEmail(value);
+      if (isEmailError) {
+        return isEmailError;
+      }
+
+      if (MD5(value).toString() !== this.props.hashedAccountEmail) {
```

Comment:
hashes are case sensitive. Convert this to lowercase (like we do [here](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L382))